contents/csharp/IntroductionToCSharp.md

```diff
@@ -0,0 +1,186 @@
+as compared to calling the actual constructor.
+
+```csharp
+public class Foo {
+    Bar MyBar;
+}
+
+Bar bar = new Bar();
+Foo foo = new Foo() {
+    MyBar = bar;
+};
+```
+
+### Closures
+
+Sometimes, it may be useful to [defer execution](http://www.informit.com/articles/article.aspx?p=2171751 "Code that is executed only when results need to be evaluated. There are many reasons for executing code later") 
+or capture a local context for later execution. Context capturing is reflected below:
+
+```csharp
+//Capturing local context
+public class Context {
+    public static Func<Int> GetCounter() {
+        int count = 0;
+        return () => {
+            count++;
+            return count();
+        }
+    }
+}
+
+Func<Int> counter = Context.GetCounter();
+counter();
+```
+
+### Nullable type
+
+Often when dealing with possible sites of [null pointer](https://en.wikipedia.org/wiki/Null_pointer "In computing, a null pointer or null reference has a value reserved for indicating that the pointer or reference does not refer to a valid object.") 
+access, many if checks might be used. However, when dealing with nullable types, such code can be shortened
+while remaining easy to read, using collaescing operators `??`, or null conditional operators `.?`. Some may find this similar to `optionals` in Swift.
+
+More can be read about Nullables [here](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/nullable-types/)
+
+```csharp
+public Car? ManufactureCar() {
+    return (hasError) ? null : new Car(param1, param2);
+}
+
+Car? car = ManufactureCar();
+//Call Drive method of car if not null
+car?.Drive();
+
+//Get amount of fuel left
+Double fuelLeft = car?.GetFuel() ?? 0;
+//Explanation:
+//If car is not null, one can expect the statement to evaluate to car.GetFuel().
+//If car is null, car?.GetFuel() evaluates to null
+//?? operator sees null, so the entire expression defaults to 0.
+```
+
+### Default Interface Implementations
+
+Sometimes classes have a common interface but do not have a common ancestor and require the same method implementation.
+A default implementation can be included in the interface to promote code reuse. Similarly, a certain class may not require certain methods listed in the interface.
+An example is a stub class that is required to implement an interface. A default method can be created with the `NotImplementedException`.
+
+The following links below describes in greater detail of some exciting features that can make your code more pleasant to read and write, such as `async/await`, `delegates` etc:
+* https://codeaddiction.net/articles/15/10-features-in-c-that-you-really-should-learn-and-use
+* https://stackify.com/csharp-8-features/
+* https://www.developer.com/net/csharp/slideshows/top-10-csharp-6.0-language-features.html
```

Comment:
Similarly, this point need to be explained better, in the article itself (i.e., don't depend on the external links to explain the point).